### PR TITLE
Upgrade Python syntax and test asserts, fix inspection and flake8

### PR DESCRIPTION
**Pull request by hugovk**
See commit messages for more info.

### CODE REVIEW

File: sourced/ml/cmd/args.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def add_feature_args(my_parser: argparse.ArgumentParser, required=True):
     for ex in extractors.__extractors__.values():
         for opt, val in ex.OPTS.items():
             my_parser.add_argument(
-                "--%s-%s" % (ex.NAME, opt), default=val, type=json.loads,
+                "--{}-{}".format(ex.NAME, opt), default=val, type=json.loads,
```
Comment (vmarkovtsev) [inline sourced/ml/cmd/args.py]: According to our [style guide](https://github.com/src-d/guide/blob/master/engineering/conventions-python.md#code-style) we do not use `format()`.

Comment (hugovk) [reply sourced/ml/cmd/args.py]: Thanks, reverted!

**Comment by vmarkovtsev** [comment]
Travis job [hanged](https://travis-ci.org/src-d/ml/jobs/438532540) - all checks passed actually.

**Comment by vmarkovtsev** [comment]
Thanks again @hugovk

**Pull request merged**